src/python/install.sh

```diff
@@ -432,18 +722,26 @@ if [[ "${INSTALL_PYTHON_TOOLS}" = "true" ]] && [[ $(python --version) != "" ]];
     find "${PIPX_HOME}" -type d -print0 | xargs -0 -n 1 chmod g+s
 
     # Update pip if not using os provided python
-    if [[ $(python --version) != "" ]] || [[ ${PYTHON_VERSION} != "os-provided" ]] && [[ ${PYTHON_VERSION} != "system" ]] && [[ ${PYTHON_VERSION} != "none" ]]; then
+    if [[ -n "${PYTHON_SRC}" ]] && [[ ${PYTHON_VERSION} != "os-provided" ]] && [[ ${PYTHON_VERSION} != "system" ]] && [[ ${PYTHON_VERSION} != "none" ]]; then
         echo "Updating pip..."
-        python -m pip install --no-cache-dir --upgrade pip
+        ${PYTHON_SRC} -m pip install --no-cache-dir --upgrade pip
     fi
 
     # Install tools
-    echo "Installing Python tools..."
     export PYTHONUSERBASE=/tmp/pip-tmp
     export PIP_CACHE_DIR=/tmp/pip-tmp/cache
     PIPX_DIR=""
     if ! type pipx > /dev/null 2>&1; then
-        pip3 install --disable-pip-version-check --no-cache-dir --user pipx 2>&1
+        
+        # python install is marked as externally managed, we need to pass
+        # pip a "--break-system-packages" option to avoid errors... This does not
+        # really breack system packages dueto the setting of PYTHONUSERBASE above,
+        # but does get us past checks for installing python packages into the
+        # system python install.
+        if python_is_externally_managed ${PYTHON_SRC}; then
+            break_system_packages="--break-system-packages"
+        fi
+        pip3 install ${break_system_packages} --disable-pip-version-check --no-cache-dir --user pipx 2>&1
```

Comment:
Using `break_system_packages` seems a bit scary to me.

Can we instead install `pipx` with check_packages (aka pkg managers)?

Reply:
We can certainly install pipx via the pkg manager if the python install is managed.

Reply:
Yes.

Reply:
Commit [baf1568](https://github.com/devcontainers/features/pull/830/commits/baf1568d80bb6f1eb73dd7e1b774b2fa5f3b662d) addresses this.